component/stream/read_write.go

```diff
@@ -47,6 +48,8 @@ import (
 	"github.com/pbnjay/memory"
 )
 
+var file_created = false
```

Comment:
this is global variable, if we execute multiple file-open/file-create in parallel then this will not work.